### PR TITLE
Added support for custom tiles

### PR DESCRIPTION
**Pull request by quantosapplications**

### CODE REVIEW

**Comment by quantosapplications** [comment]
The second commit is for adding crs simple support

File: lib/src/geo/crs/crs.dart [hunk 1]
```diff
@@ -95,6 +121,34 @@ abstract class Projection {
...
+	
+    @override
+	  LatLng unproject(CustomPoint point) {
+	    return LatLng(point.y, point.x);
+	  }
+	
+	  /*@override
+	  void setBounds(Bounds<double> bounds) {
+	    _bounds = bounds;
+	  }*/
```
Comment (johnpryan) [inline lib/src/geo/crs/crs.dart]: please remove this code

File: lib/src/layer/tile_layer.dart [hunk 2]
```diff
@@ -92,6 +92,9 @@ class TileLayerOptions extends LayerOptions {
   final int keepBuffer;
   ImageProvider placeholderImage;
   Map<String, String> additionalOptions;
+  
+  //function to create custom tile
```
Comment (johnpryan) [inline lib/src/layer/tile_layer.dart]: please add a space after `//` and capitalize the first word

File: lib/src/geo/crs/crs.dart [hunk 3]
```diff
@@ -56,6 +56,32 @@ abstract class Crs {
   Tuple2<double, double> get wrapLat;
 }
 
+///Custom CRS for non geographical maps
```
Comment (johnpryan) [inline lib/src/geo/crs/crs.dart]: please add a space after `///`

File: lib/src/geo/crs/crs.dart [hunk 4]
```diff
@@ -56,6 +56,32 @@ abstract class Crs {
...
+	class CrsSimple extends Crs {
+    @override
+	  final String code = 'CRS.SIMPLE';
+
+    @override
+	  final Projection projection;
+
+    @override
+	  final Transformation transformation;
```
Comment (johnpryan) [inline lib/src/geo/crs/crs.dart]: The formatting looks weird here. did you run `dartfmt` on this file?

File: lib/src/geo/crs/crs.dart [hunk 5]
```diff
@@ -95,6 +121,34 @@ abstract class Projection {
   LatLng unproject(CustomPoint point);
 }
 
+
+class LonLat extends Projection {
```
Comment (johnpryan) [inline lib/src/geo/crs/crs.dart]: What is this class? Is there a better name we can use?

**Comment by johnpryan** [comment]
LGTM - thanks for the contribution!

**Comment by johnpryan** [comment]
After looking at this some more, is there a reason why you can't use your own tileProvider instead?

**Pull request merged**